drivers/input/input_kbd_matrix.c

```diff
@@ -313,9 +334,33 @@ static void input_kbd_matrix_polling_thread(void *arg1, void *unused2, void *unu
 	}
 }
 
+#ifdef CONFIG_PM_DEVICE
+int input_kbd_matrix_pm_action(const struct device *dev,
+			       enum pm_device_action action)
+{
+	struct input_kbd_matrix_common_data *data = dev->data;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+		atomic_set(&data->suspended, 1);
+		break;
+	case PM_DEVICE_ACTION_RESUME:
+		atomic_set(&data->suspended, 0);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	input_kbd_matrix_poll_start(dev);
```

Comment:
Not sure I follow, there should be 4 possible combinations
- suspend, scanning cycle idle: the scanning cycle will wake up, run a cycle while reading all rows as 0 (because the suspended bit is set) then fall back on the polling lock with detect mode deactivated and stay idle from there in `input_kbd_matrix_polling_thread`
- suspend, scanning cycle running (keys pressed): the cycle will start reading all rows as 0 (suspend bit set), report a release of all the keys, eventually also fall back in `input_kbd_matrix_polling_thread`, find the lock unlock, do another dummy round, then fall back again and suspend for good
- resume, driver idle and suspended (i.e. waiting in `input_kbd_matrix_polling_thread`), just start scanning normally
- resum, driver running: the cycle will start reading the actual rows value and emitting keys, eventually fall back to `input_kbd_matrix_polling_thread`, do another dummy round, lock again but this time with detection mode enabled

Does it make sense? There's some space for optimizing it but I think it should cover all cases.

Reply:
Acknowledged, other descriptions are as follows.